HW02/Ivan_Papushoi/ivan_papushoi_hw_02_t_03.rb

```diff
@@ -0,0 +1,77 @@
+# method should return a string containing the duration of the action
+# in SECONDS between events that occurred at the time
+# of Calling core with action:
+# between them can be any number of other events.
+
+def receive_filelines
+  log_file = File.open(FILE_NAME)
+  file_lines = log_file.readlines
+  log_file.close
+  file_lines
+end
+
+def receive_cores(file_lines)
+  file_lines.select { |line| line.downcase.include?('calling core') }
+end
+
+def parse(lines_call_core)
+  reg_exp_date_time = /\d{4}\-\d{2}\-\d{2} \d{2}\:\d{2}\:\d{2}\.\d{1}/
+  lines_call_core.map { |line| line.match(reg_exp_date_time)[0] }
+end
+
+def catch_ymd(record)
+  y_m_d = []
+  year = record.match(/\d{4}/)[0]
+  month = record.match(/\-\d{2}\-/)[0].delete '-'
+  day = record.match(/\-\d{2} /)[0].strip.delete '-'
+  y_m_d << year << month << day
+  y_m_d
+end
+
+def catch_hms(record)
+  h_m_s = []
+  hour = record.match(/ \d{2}/)[0].strip
+  minute = record.match(/:\d{2}\:/)[0].delete ':'
+  second = record.match(/\:\d{2}\.\d{1}/)[0].delete ':'
+  h_m_s << hour << minute << second
+  h_m_s
+end
+
+def _to_nums(catched_array)
```

Comment:
Why do you use such a strange prefix?
```suggestion
def to_nums(catched_array)
```

Reply:
Delete that strange prefix :smile: